src/playlist.js

```diff
@@ -91,7 +107,8 @@
     // available PTS information
     for (left = range.start; left < range.end; left++) {
       segment = playlist.segments[left];
-      if (segment.minVideoPts !== undefined) {
+      if (segment.minVideoPts !== undefined ||
```

Comment:
The entries of a media playlist should be segment objects-- `[undefined]` is not a legal value for that array. Side note: it would be better to file this sort of thing as an issue. Line comments might get overlooked.